apiserver/application/application_unit_test.go

```diff
@@ -252,11 +316,175 @@ func (s *ApplicationSuite) TestDestroyUnit(c *gc.C) {
 	}})
 }
 
+func (s *ApplicationSuite) setUpSeriesSubordinates() {
+	ch := mockCharm{
+		meta: &charm.Meta{
+			Series: []string{"trusty", "xenial"},
+		},
+	}
+	app := mockApplication{
+		charm:       &ch,
+		charmURL:    "cs:foo-sub",
+		subordinate: true,
+		units: []mockUnit{{
+			tag: names.NewUnitTag("foo-sub/0"),
+		}, {
+			tag: names.NewUnitTag("foo-sub/1"),
+		}},
+	}
+	s.charms["cs:foo-sub"] = &ch
+	s.applications["foo-sub"] = &app
+	s.backend.charms["cs:foo-sub"] = &ch
+	s.backend.applications["foo-sub"] = &app
+	for i, u := range s.applications["foo"].units {
+		u.subordinateNames = []string{fmt.Sprintf("foo-sub/%d", i)}
+	}
+}
+
+func (s *ApplicationSuite) TestApplicationUpdateSeries(c *gc.C) {
+	s.setUpBasicTest()
+	args := params.ApplicationUpdateSeriesArgs{
+		Args: []params.ApplicationUpdateSeriesArg{{
+			ApplicationName: "foo",
+			Series:          "trusty",
+		}, {
+			ApplicationName: "foo",
+			Series:          "zesty",
+		}},
+	}
+	results, err := s.api.UpdateApplicationSeries(args)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(len(results.Results), gc.Equals, 2)
+	c.Assert(results.Results[0], jc.DeepEquals, params.ErrorResult{})
+	c.Assert(results.Results[1], jc.DeepEquals, params.ErrorResult{
+		Error: &params.Error{
+			Code:    params.CodeIncompatibleSeries,
+			Message: `"zesty" is incompatible`,
```

Comment:
this message is better as something aligned with the existing message used when the user attempt to deploy an app with an incompatible series